src/test/compile-fail-fulldeps/proc-macro/feature-gate-proc_macro.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// aux-build:attr_proc_macro.rs
+// gate-test-proc_macro
+#![feature(use_extern_macros)]
+
+extern crate attr_proc_macro;
+use attr_proc_macro::attr_proc_macro;
+
+//~ ERROR: attribute procedural macros are experimental
+#[attr_proc_macro]
```

Comment:
Currently, `check_proc_macro_attrs` only checks `builtin_macros`, so it will only work with `#[macro_use]`-imported proc-macros.

To support the `use` imports in this test (which need `#![feature(use_extern_macros)]`), we would have to check
```rust
self.resolve_ident_in_lexical_scope(Ident::with_empty_ctxt(name), MacroNS, None)
```
in addition to `self.builtin_macros.get(&name)`.